meshpy/FourC/beam_potential.py

```diff
@@ -0,0 +1,256 @@
+            Functions for line charge densities of a potential law in form of a
+            power law. Same number of functions and prefactors/exponents/line
+            charge densities must be provided!
+        """
+
+        self.input_file = input_file
+
+        # if only one potential law prefactor/exponent is present, convert it
+        # into a list for simplified usage
+        if isinstance(pot_law_prefactor, (float, int)):
+            pot_law_prefactor = [pot_law_prefactor]
+        if isinstance(pot_law_exponent, (float, int)):
+            pot_law_exponent = [pot_law_exponent]
+        if isinstance(pot_law_line_charge_density, (float, int)):
+            pot_law_line_charge_density = [pot_law_line_charge_density]
+
+        # check if same number of prefactors and exponents are provided
+        if (
+            not len(pot_law_prefactor)
+            == len(pot_law_exponent)
+            == len(pot_law_line_charge_density)
+        ):
+            raise ValueError(
+                "Number of potential law prefactors do not match potential law exponents!"
+            )
+
+        self.pot_law_prefactor = pot_law_prefactor
+        self.pot_law_exponent = pot_law_exponent
+        self.pot_law_line_charge_density = pot_law_line_charge_density
+        self.pot_law_line_charge_density_funcs = pot_law_line_charge_density_funcs
+
+    def add_header(
+        self,
+        *,
+        potential_type="Volume",
+        cutoff_radius=None,
+        evaluation_strategy=None,
+        regularization_type=None,
+        regularization_separation=None,
+        integration_segments=1,
+        gauss_points=10,
+        potential_reduction_length=-1,
+        automatic_differentiation=False,
+        choice_master_slave=None,
+        option_overwrite=False,
+    ):
+        """
+        Set the basic header options for beam potential interactions.
+
+        Args
+        ----
+        potential_type: string
+            Type of applied potential (volume, surface).
+        cutoff_radius: float
+            Neglect all contributions at separation larger than this cutoff
+            radius.
+        evaluation_strategy: string
+            Strategy to evaluate interaction potential.
+        regularization_type: string
+            Type of regularization to use for force law at separations below
+            specified separation (constant_extrapolation, linear_extrapolation).
+        regularization_separation: float
+            Use specified regularization type for separations smaller than
+            this value.
+        integration_segments: int
+            Number of integration segments to be used per beam element.
+        gauss_points: int
+            Number of Gauss points to be used per integration segment.
+        potential_reduction_length: float
+            Potential is smoothly decreased within this length when using the
+            singlelengthsspecific approach to enable an axial pull off force.
```

Comment:
Is `singlelengthsspecific` a well know word in the beam potential framework?

Reply:
I would say so, a big differentiation between single length specific (SBIP) and double length specific (SSIP) methods is possible.

Reply:
Ok, can you then change the documentation to something like:
```
Potential is smoothly decreased within this length when using the
single length specific (SBIP) approach to enable an axial pull off force.
```
The compound word `singlelengthsspecific` looks weird / like a typo in the documentation.